course/steps/06_graphql-implementation/en.md

```diff
@@ -0,0 +1,106 @@
+# GraphQL: Retrieving data from Masterdata
+
+## Introduction
+
+Now that we have updated the products count, we need to retrieve the _top n_ most viewed products. We can, then, use Masterdata to retrieve the product page view data and sort by the _count_ field. We can also limit the length of retrieved products, creating a customized size rank of most visited products.
+
+## GraphQL
+
+To get these product page views, we will use [GraphQL](https://graphql.org/), the technology used by VTEX IO for data fetching, to implement a query to Masterdata. GraphQL allows us to implement queries in a simple and easy way, specifying the data you want to retrieve. This makes your API reliable, since GraphQL controlls the data fetched instead of the server itself.
+
+Therefore, GraphQL uses types and a query schema to specify the data retrieved and resolvers to get the exact data needed.
+
+## Activity
+
+1. On the directory `/graphql/types` create the `productView.graphql` file and declare the type of the product list we want to retrieve:
+
+    ```
+    type ProductView {
+        slug: String
+        count: Int
+    }
+    ```
+
+2. Still in the `/graphql` directory, define the schema in the `schema.graphql` file:
+
+    ```
+    type Query {
+        productList(topN: Int): [ProductView]
+    }
+    ```
+
+    > Keep in mind that the schema will define the structure of our query and the retrieved data.
```

Comment:
Adiciona aqui alguma coisa sobre as diretivas que dá pra usar https://github.com/vtex-apps/service-example

Isso é importante porque se for algo que precisar do token do usuário ou de algum cookie (como o OrderForm) tem que ser assim

Reply:
Esse é o link mesmo? 🤔

Reply:
Porém acho super válido o que ce falou 😬

Reply:
Boa, é https://github.com/vtex-apps/graphql-example